### PR TITLE
Addition of ISO Standard 11855-2 Modeling of Fluid to Slab Heat Transfer in Hydronic Low Temperature Radiant Systems

### LINKED ISSUE #8056: Addition of Pipe and Pipe to Slab Contact Resistance to Low Temperature Radiant Models
Issue overview
--------------
The low temperature radiant system model in EnergyPlus ignores any thermal resistance due to the presence of the pipe material for the tubing that runs through the system with a heat transfer fluid being circulated to provide heating and cooling.  It also ignores any contact resistance between the pipe and the slab.  The thought of the initial implementation was that the main driver of heat transfer was the fluid to solid heat exchange.  Recent work has shown that there is perhaps a good reason to include the pipe material and pipe to slab contact resistance in the model.  So, this work (requested by the Center for the Built Environment, University of California Berkeley based on their expertise with radiant systems) is to implement the addition of pipe thermal resistance and pipe to slab contact resistance to the EnergyPlus model.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): all
 - Version of EnergyPlus (if using an intermediate build, include SHA): current
 - Unmethours link or helpdesk ticket number: none, requested new feature by CBE.

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [X] Defect file added (list location of defect file here): none, new feature
 - [X] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by RKStrand**
Pull request overview
---------------------
 - Fixes #8056 
 - This pull request addresses another low temperature radiant system enhancement requested by the Center for the Built Environment (CBE) at the University of California Berkeley.  This enhancement adds a new model for characterizing the heat transfer between the fluid being circulated through the hydronic (constant and variable flow) low temperature radiant system models in EnergyPlus.  In the legacy model integrated into EnergyPlus from IBLAST (a research version of BLAST), it was assumed that convection between the fluid and the inside of the tubing was the main mode of heat transfer and modeling only convection was sufficient due to approximations made at the location of the heat source/sink.  However, historically, this assumption has perhaps been too simplistic and often leads to a temperature at the source/sink location that is equal to the outlet water conditions.  In addition, there exists in ISO Standard 11855-2 (2012) entitled "Building environment design — Design, dimensioning, installation and control of embedded radiant heating and cooling systems —  Part 2: Determination of the design heating and cooling capacity" a method for taking the conduction through the tube itself into account.  This pull request implements the ISO Standard 11855-2 method as an option for the user to choose as an alternative to the existing convection only model.  Two new input files were added to show how the new modeling capability works.  Two new unit tests were also added.

All existing EnergyPlus test suite files will continue to use the old method for calculating heat transfer between the fluid and the system.  So, the only differences in the results should be some audit file differences because the number of parameters with default values has increased.  There should be no differences in the actual program output for any file.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [X] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [X] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [none expected] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [NA] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [NA] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [x] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [NA] If structural output changes, add to output rules file and add OutputChange label
 - [NA] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] ~If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix~
 - [x] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [x] Check any new function arguments for performance impacts
 - [x] Verify IDF naming conventions and styles, memos and notes and defaults
 - [x] If new idf included, locally check the err file and other outputs
 - [x] If doc changes, review pdfs.
 - [x] If transition required, test.

### CODE REVIEW

File: doc/engineering-reference/src/simulation-models-encyclopedic-reference-004/radiant-system-models.tex [hunk 1]
```diff
@@ -672,6 +675,40 @@ \subsubsection{Heat Exchanger Formulation for Hydronic Systems}\label{heat-excha
 
 The parameter m is the absolute viscosity of water.~ For internal pipe flow, the flow is assumed to be turbulent for ReD \(\geq\) 2300.
 
+\textbf{\emph{ISOStandard Heat Transfer Model}}.  In this model, the tubes and contact resistance are assumed to have an impact on the overall heat transfer.  So, conduction through the tube wall and the effect of an imperfect connection between the pipe itself and the surrounding system must be determined in addition to convection.  The method used in EnergyPlus for this model is based on ISO Standard 11855-2(2012) which is entitled "Building environment design — Design, dimensioning, installation and control of embedded radiant heating and cooling systems — Part 2: Determination of the design heating and cooling capacity".  The key equations from the ISO Standard are summarized in Appendix B of Standard 11855-2 and are outlined below.
```
Comment (mjwitte) [inline doc/engineering-reference/src/simulation-models-encyclopedic-reference-004/radiant-system-models.tex]: LaTeX smart quotes are preferred. ` ``quoted text'' ` Here and below.

File: idd/Energy+.idd.in [hunk 2]
```diff
@@ -44181,17 +44205,42 @@ ZoneHVAC:LowTemperatureRadiant:ConstantFlow,
...
    N1 , \field Hydronic Tubing Inside Diameter
         \units m
         \minimum> 0
         \default 0.013
-   N2 , \field Hydronic Tubing Length
-        \note Total length of pipe embedded in surface
+        \ip-units in
+   N2 , \field Hydronic Tubing Outside Diameter
+        \units m
+        \minimum>0
```
Comment (mjwitte) [inline idd/Energy+.idd.in:44229]: Need a space after >

**Review by mjwitte — COMMENTED** [review]
A couple of initial comments.

File: doc/engineering-reference/src/simulation-models-encyclopedic-reference-004/radiant-system-models.tex (hunk 1 above)
Comment (RKStrand) [reply doc/engineering-reference/src/simulation-models-encyclopedic-reference-004/radiant-system-models.tex]: Got it--will fix this and the other issue below and will make another commit, probably sometime tomorrow AM.

File: idd/Energy+.idd.in (hunk 2 above)
Comment (RKStrand) [reply idd/Energy+.idd.in:44229]: Sigh--not sure how that go in there, but I'll fix that.  Should show up in the next commit.

Comment (mjwitte) [reply idd/Energy+.idd.in:44229]: Same for the other object.

File: testfiles/RadLoHydrHeatCoolAuto.idf [hunk 3]
```diff
@@ -1871,8 +1871,11 @@
     RadiantSysAvailSched,    !- Availability Schedule Name
     West Zone,               !- Zone Name
     Zn001:Flr001,            !- Surface Name or Radiant Surface Group Name
+    ConvectionOnly,          !- Fluid to Radiant Surface Heat Transfer Model
     0.013,                   !- Hydronic Tubing Inside Diameter {m}
+    0.016,                   !- Hydronic Tubing Inside Diameter {m}
```
Comment (mjwitte) [inline testfiles/RadLoHydrHeatCoolAuto.idf]: @RKStrand Also, in all(?) of the testfiles, the field name comments for Outside Diameter all say Inside Diameter.

**Review by mjwitte — COMMENTED** [review]
Ran all unit test serially, all pass. 

Ran RadLoTempCFloHeatCool.idf and RadLoTempCFloISOPipeHT.idf as annual simulations with Chicago weather. Electricity use was about the same, shifting from cooling electricity to more pumping electricity. But DistrictHeating dropped from 142 to 107 GJ. Is that correct? Also showing large changes (6x) in Peak Cooling Sensible Heat Gain Components "HVAC Input Cooled Surface Cooling [W]" (similar for heating).

Similar comparison for variable flow show small change in District Heating and large change in pump power (from 9 to 22 GJ with ISO model).

File: testfiles/RadLoHydrHeatCoolAuto.idf (hunk 3 above)
Comment (RKStrand) [reply testfiles/RadLoHydrHeatCoolAuto.idf]: Clearly a bad cut and paste on my part. 🤦‍♂️ I'll fix that in the next commit.

**Comment by RKStrand** [comment]
@mjwitte I'll look at the RadLoTempCFloHeatCool.idf and RadLoTempCFloISOPipeHT.idf results for annual runs and get back to you.  I suspect that there might be some issues with the condensation controls that are causing this "weirdness".  From what I saw in my runs, condensation should be much more likely in the ConvectionOnly model because of the much lower temperature prediction at the source location and thus at the floor surface temperature.  As a result, the ConvectionOnly runs only kick in every once in a while and then really do a bunch of cooling while the ISOStandard stays on more consistently.  That's actually good news though because the ConvectionOnly generally leads to a lot of shutdown due to condensation.  The way to check on that is to do an annual run with the condensation controls turned off to see what will happen.

Either way, I would expect the peak heating and cooling numbers to be lower because with the lower delta T on the water loop you should get less heating/cooling, but I think it will actually be a much more realistic number with the ISOStandard method.  With the ConvectionOnly model, the water exit temperature was almost always equal to the temperature at the source location.  While that is the limit and doesn't violate any laws of thermodynamics, it's always been a bit optimistic.  So, it's a good thing that the ISOStandard method is being implemented.  I'll check this locally on my machine by modifying the input files so that condensation controls are OFF for all systems and then post again here.  With the condensation controls off, we might see more "reasonable" changes.  From what I have seen though with the exit temperatures and the source location temperatures, the new calculation procedure makes sense and is more believable than the ConvectionOnly.

**Comment by RKStrand** [comment]
@mjwitte Ok, so I have an answer.  With the vastly different reactions at the source location because of the big difference in how much heat transfer there is happening at that point, the ISOStandard results in less heat transfer because in general the temperature differences are smaller.  That's to be expected because of the reduced heat transfer in the ISOStandard.  Looking at some of the results for the annual run for the constant flow file, this is born out.  The delta T for the ISOStandard file is always less--exactly what we would expect.  So, the loads (max and sum) are less.  But...keep in mind that the actual zone results are vastly different.  If you look at the Fanger PMV, you will also see that the PMV for the ISOStandard method is worse by quite a bit.  So, the smaller maximum load and sums are a bit of a red herring because the thermal comfort is not equivalent (not even close).  Could we get it close?  Well, yes if we had PMV control for the radiant systems which we don't (maybe in the future this will be proposed and accepted as a future task?).  I'm dropping in a file that has some results to compare things a little easier rather than in two separate files.

[LowTempRadCFloCompare20200811.xlsx](https://github.com/NREL/EnergyPlus/files/5058836/LowTempRadCFloCompare20200811.xlsx)

Does this help explain things a bit?  Basically, what someone would need to do in an actual situation is go back and update the controls for the system to come closer to equivalent comfort based on the fact that the temperatures in the space (particularly the floor temperature) are now significantly different and comfort will be different as well.  Changing the controls could get things back to closer PMVs and then I think you would have similar results for max load and the load sum.

**Comment by RKStrand** [comment]
@mjwitte I talked about the constant flow system but failed to comment on what you saw for the variable flow.  I think the variable flow will attempt to increase the flow to try to get more heat transfer.  So, the fact that the flow increased makes sense and with the much lower delta T it doesn't surprise me that the pump energy increase is fairly significant.  I'll double check that and post again.

**Comment by mjwitte** [comment]
@RKStrand A little, but it still doesn't add up. The system is controlling on MeanAirTemperature, so that should be the same (or very close) using the two heat transfer models. For the ISO case the zone air temperature is roughly 2C below the ConvectionOnly case when the radiant floor is actively heating.

**Comment by RKStrand** [comment]
@mjwitte I think that is because the ISO simply doesn't have the capacity (flow) so it can't meet the setpoint.  So, in the file I posted, look at line 255 when max heating happens for that time frame for the ConvOnly.  The delta T (Q255-R255) is around 10C.  The max heating for the ISOStd happens on line 258 where delta T (AA258-AB258) is only about 3.4.  Because the flow rates are the same (0.3908) the load has to be different by around that same factor (10/3.4 or around 3).  The ISOStd version is trying to catch up (see the fact that the ISOStd temperatures are higher meaning that it's further away from the goal and trying to catch up) but it simply doesn't have the flow to equal what is happening in the ConvOnly result.  In fact, that's another indication that the MAT is not the same between the two runs--if it was, the inlet temperature to the radiant system (see columns Q vs. AA) would be similar also.  I'll try a run with a higher flow rate and see what happens.

**Comment by RKStrand** [comment]
@mjwitte Here's what I tried.  Tripled the flow rate--didn't change much other than cutting the delta T by a factor of 3.  Switched flow back to original and changed the temperature schedules by 5C (roughly the difference between the source location temps in the previous attempts and the ConvOnly).  So, heating water temps vary from 35 to 55C (previously from 30-50C).  Made a slight dent but still quite a ways off.

So, I went radical.  Kept the 5C temperature change in the water schedules but also temporarily eliminated the contact resistance.  My thought: the pipe resistance is definitely higher than the equivalent convection resistance, but the contact resistance is at least four times higher than the pipe resistance for this example according to the ISO standard equation.  Is that really reasonable?  For a pipe that is embedded in concrete?  Results: very similar to the ConvOnly case.  Source location temperatures are pretty close.  PMVs are pretty close.  Sum of the heating loads around 2% different (higher for this case but slightly better PMV balances that out).  Peak is about 30% smaller for this test case, but I think because the lack of resistance allows the ConvOnly to respond faster and it ends up with a "pick-up" load rather a smoother response from the Conv+Pipe.

What do we make of all of this?  Well, it's possible that the equation for contact resistance is wrong OR that it isn't intended as that.  The language in the standard is a bit confusing and could be interpreted different ways potentially.  CBE clearly felt this was an equation for contact resistance.  But the contact resistance equation is a function of the distance between the adjacent pipes.  So, was that equation supposed to be contact resistance or resistance between the pipes?  Why would the contact resistance change with pipe separation and increase when the pipe separation increases?  I'm going to reach out to CBE and see if they see anything wrong with this logic.

What could we do in the meantime?  Well, I supposed I could implement two options--one with the convection and pipe resistance as per the ISO and another with that plus the contact resistance.  What do you think of all of this?

**Comment by RKStrand** [comment]
@mjwitte Ok, I went back and read the ISO standard more carefully (like I should have to begin with 🤦‍♂️).  The "contact resistance" is not contact resistance--it's essentially to get from the outer surface of the pipe to the "average" temperature at that depth within the slab.  In other words, it allows the model to sort of take 2-d effects into account and then model conduction as 1-D beyond that.  That's great but it doesn't apply to the E+ model.  So, I think what needs to happen is the "contact resistance" that really isn't contact resistance needs to be ripped out of the model, everything needs to be adjusted for that, and the other changes you recommended need to work.  I will also update the new constant flow IDF file to have different water temperatures so that the file it most closely resembles will end up with fairly similar results.  I'll also check on the variable flow file too.  I'd like to hear back from CBE, but given the upcoming deadline and the fact that I'm pretty certain I have understood things correctly, I'm going to move ahead on that basis.  Thanks for reviewing this and causing me to re-examine the results.  This would have been an incorrect implementation otherwise.

**Comment by mjwitte** [comment]
@RKStrand Good to hear. If necessary, we can merge this before you hear back from CBE. If needed, we can post a follow-up issue to make any corrections to the algorithm after I/O Freeze as long as we don't need any further IDD changes.

**Comment by RKStrand** [comment]
Thanks, @mjwitte.  The only thing I can think of is that CBE would come back and say that we need a different equation for contact resistance and I don't think that will be the case.  The way radiant systems are modeled in E+ means that equation that I misinterpreted as "contact resistance" is already taken into account so it's not needed.  The pipe resistance is still a good addition, and I think it will improve the model.  In any case, I'm moving forward with my revised plan.  Will take a bit of extra work, but it's the correct implementation and I'd rather have it right.  Next commit might not hit until tomorrow AM, but I'll let you know when it's ready.  Thanks again!

**Comment by RKStrand** [comment]
@mjwitte Ok, things are looking better now.  I'm attaching another summary spreadsheet here.  This is all without the "contact resistance" that wasn't actually contact resistance for the ISO Standard model.

[LowTempRadCompConvOnlyISOupdated.xlsx](https://github.com/NREL/EnergyPlus/files/5060019/LowTempRadCompConvOnlyISOupdated.xlsx)

Diffs in sum values are largely due to diffs in PMV.  Diffs in max values are due I believe largely to how fast the ConvOnly model can pump heat into or out of a slab.  For the variable flow, I plan to modify the new IDF to include a change in the CHW temp as shown in the last example results.  While the results in this spreadsheet are all for the condensation controls turned off for all zones, I won't change the input for the existing input files because those actually are the tests for condensation controls.  If I turn them all off, we won't have any tests of condensation controls.

Hope to check in revised versions of all files tomorrow AM.  If you feel like things are still not quite right, let me know and we can discuss that further.

**Comment by RKStrand** [comment]
@mjwitte Got to it sooner than I thought.  This will give it overnight to run all the CI test.  Let me know if there are further concerns.  Thanks!

File: doc/engineering-reference/src/simulation-models-encyclopedic-reference-004/radiant-system-models.tex [hunk 4]
```diff
@@ -672,6 +675,32 @@ \subsubsection{Heat Exchanger Formulation for Hydronic Systems}\label{heat-excha
 
 The parameter m is the absolute viscosity of water.~ For internal pipe flow, the flow is assumed to be turbulent for ReD \(\geq\) 2300.
 
+\textbf{\emph{ISOStandard Heat Transfer Model}}.  In this model, the convection between the fluid and the tubes and the conduction resistance of the tubes themselves are assumed to have an impact on the overall heat transfer.  The method used in EnergyPlus for this model is based on ISO Standard 11855-2(2012) which is entitled``Building environment design — Design, dimensioning, installation and control of embedded radiant heating and cooling systems — Part 2: Determination of the design heating and cooling capacity''.  The key equations from the ISO Standard are summarized in Appendix B of Standard 11855-2 and are outlined below.
```
Comment (mjwitte) [inline doc/engineering-reference/src/simulation-models-encyclopedic-reference-004/radiant-system-models.tex]: If any other changes are required - need a space here after entitled.

File: doc/input-output-reference/src/overview/group-radiative-convective-units.tex [hunk 5]
```diff
@@ -1161,8 +1180,11 @@ \subsubsection{Inputs}\label{inputs-5-022}
       RADIANTSYSAVAILSCHED,    !- Availability Schedule Name
       SPACE1-1,                !- Zone Name
       C1-1,                    !- Surface Name or Radiant Surface Group Name
+      ConvectionOnly,          !- Fluid to Radiant Surface Heat Transfer Model
       0.013,                   !- Hydronic Tubing Inside Diameter {m}
+      0.016,                   !- Hydronic Tubing Inside Diameter {m}
```
Comment (mjwitte) [inline doc/input-output-reference/src/overview/group-radiative-convective-units.tex]: Inside --> Outside

File: doc/input-output-reference/src/overview/group-radiative-convective-units.tex [hunk 6]
```diff
@@ -1438,8 +1480,11 @@ \subsubsection{Inputs}\label{inputs-6-019}
       RadiantSysAvailSched,   !- availability schedule
       Resistive Zone,         !- Zone name
       Zn001:Flr001,           !- Surface name or group
-      0.012,                  !- Hydronic tubing inside diameter {m}
-      400.0,                  !- Hydronic tubing length {m}
+      ConvectionOnly,         !- Fluid to Radiant Surface Heat Transfer Model
+      0.013,                  !- Hydronic Tubing Inside Diameter {m}
+      0.016,                  !- Hydronic Tubing Inside Diameter {m}
```
Comment (mjwitte) [inline doc/input-output-reference/src/overview/group-radiative-convective-units.tex]: Inside --> Outside

File: idd/Energy+.idd.in [hunk 7]
```diff
@@ -44012,18 +44012,42 @@ ZoneHVAC:LowTemperatureRadiant:VariableFlow,
...
         \object-list RadiantGroupNames
+   A5 , \field Fluid to Radiant Surface Heat Transfer Model
+        \note This parameter identifies how the heat transfer between
+        \note fluid being circulated through the radiant system and
+        \note the radiant system (slab) is modeled.  ConvectionOnly
+        \note means that only convection between the fluid and the
+        \note inside surface of the pipe is modeled.  ISOStandard
+        \note models convection between the fluid and the inside of
+        \note of the pipe, conduction through the pipe material, and
+        \note contact resistance between the outside surface of the
```
Comment (mjwitte) [inline idd/Energy+.idd.in:44023]: Remove contact resistance from here? Also, this description implies that the method for the fluid resistance is the same in both options and the ISO option simply adds the pipe resistance. But the ISO method does everything differently. Not coming up with a concise way to say this.

File: idd/Energy+.idd.in [hunk 8]
```diff
@@ -44181,17 +44205,42 @@ ZoneHVAC:LowTemperatureRadiant:ConstantFlow,
...
```
Comment (mjwitte) [inline idd/Energy+.idd.in:44216]: remove reference to contact resistance.

File: tst/EnergyPlus/unit/LowTempRadiantSystem.unit.cc [hunk 9]
```diff
@@ -495,8 +495,11 @@ TEST_F(EnergyPlusFixture, AutosizeLowTempRadiantVariableFlowTest)
         "    RadiantSysAvailSched,    !- Availability Schedule Name",
         "    West Zone,               !- Zone Name",
         "    Zn001:Flr001,            !- Surface Name or Radiant Surface Group Name",
+        "    ConvectionOnly,          !- Fluid to Radiant Surface Heat Transfer Model",
         "    0.012,                   !- Hydronic Tubing Inside Diameter {m}",
+        "    0.016,                   !- Hydronic Tubing Inside Diameter {m}",
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/LowTempRadiantSystem.unit.cc]: Outside diameter

**Review by mjwitte — APPROVED** [review]
Unit tests all pass.
Comparisons between ConvectionOnly and ISOStandard are much better now.
Tested transition on all RadLo* testfiles from 9.3.0 to this 9.4.0.
Just a few minor cleanups then this can go in.

File: doc/engineering-reference/src/simulation-models-encyclopedic-reference-004/radiant-system-models.tex (hunk 4 above)
Comment (RKStrand) [reply doc/engineering-reference/src/simulation-models-encyclopedic-reference-004/radiant-system-models.tex]: Fixed in next commit.

File: doc/input-output-reference/src/overview/group-radiative-convective-units.tex (hunk 5 above)
Comment (RKStrand) [reply doc/input-output-reference/src/overview/group-radiative-convective-units.tex]: Fixed in next commit.

File: doc/input-output-reference/src/overview/group-radiative-convective-units.tex (hunk 6 above)
Comment (RKStrand) [reply doc/input-output-reference/src/overview/group-radiative-convective-units.tex]: Fixed in next commit.

File: idd/Energy+.idd.in (hunk 7 above)
Comment (RKStrand) [reply idd/Energy+.idd.in:44023]: Cleaned this up--will show up in the next commit.

File: idd/Energy+.idd.in (hunk 8 above)
Comment (RKStrand) [reply idd/Energy+.idd.in:44216]: Cleaned this up--will show up in the next commit.

File: tst/EnergyPlus/unit/LowTempRadiantSystem.unit.cc (hunk 9 above)
Comment (RKStrand) [reply tst/EnergyPlus/unit/LowTempRadiantSystem.unit.cc]: Wow--I really need to be a bit more careful with my cut and pasting!  Fixed in the next commit.

**Comment by mjwitte** [comment]
@RKStrand Thanks for the final cleanup. CI is happy (except the random unit test failure on Mac, not related to this branch). Merging.

**Comment by Tokarzewski** [comment]
Where can I input su - "Thickness of the layer above the pipe" or choose system type A,B,C,D?
image

**Comment by RKStrand** [comment]
@Tokarzewski You do that using the various layers of the construction and the source after layer X in the Construction:InternalSource input.  The second field is "Source Present After Layer Number".  Split layer 2 in your diagram into two separate layers so that the mid-point of the pipes.  One of the layers will have a thickness of (Su + 0.5D) and the other layer will have a thickness that is the remaining thickness of your layer 2.  Hope this helps--if not, let me know.

**Pull request merged**